src/detectors/BTOF/BarrelTOFNeighborFinder.cc

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024 Chun Yuen Tsang, Souvik Paul, Prithwish Tribedy
+// Date: 10/15/2024
+
+#include "BarrelTOFNeighborFinder.h"
+#include "services/geometry/richgeo/RichGeo.h"
+
+// Include appropriate class headers. e.g.
+#include "TAxis.h"
+#include "TCanvas.h"
+#include "TGeoManager.h"
+
+#include <algorithm>
+#include <iostream>
+
+void BarrelTOFNeighborFinder::init(const dd4hep::Detector* detector) {
+  // you need to init the class before calling any other methods
+  // What about RAII?
+  _detector  = detector;
```

Comment:
Preferred way of getting the geometry is through the geometry service, see https://github.com/eic/EICrecon/blob/main/src/algorithms/fardetectors/FarDetectorTrackerCluster.cc#L29